### PR TITLE
added mpa-7103 - basic CRUD tests for configuration service

### PR DESCRIPTION
**Pull request by PavloEsco**
Automation testing for the configuration service - basic CRUD tests

### CODE REVIEW

File: scalecube-configuration/src/test/java/resources/features/mpa-7103-Automation-testing-for-the-configuration-servicen.feature [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+  #MPA-7103 (#2.1)
+  Scenario: Fail to save (edit) the write permission for specific entity assigned by "Member" role
+    Given a user have got a valid "token" (API key) with assigned "member" role
+    And the relevant specified name "repository" was created and stored in DB
+    When this user requested to save the "specific" entity (key and value) with write permission for the relevant "repository" specified name
+    Then this new entity (key and value) shouldn't be created and the user should get an error message: "role: permission denied"
+
+
+  #MPA-7103 (#2.2)
+  Scenario: Fail to save (edit) the write permission for specific entity which already exist (duplicate)
```
Comment (aharonha) [inline scalecube-configuration/src/test/java/resources/features/mpa-7103-Automation-testing-for-the-configuration-servicen.feature]:
```suggestion
  Scenario: Update (override - edit) the value for specific entity which already exist
```

File: scalecube-configuration/src/test/java/resources/features/mpa-7103-Automation-testing-for-the-configuration-servicen.feature [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+    Given a user have got a valid "token" (API key) with assigned "owner" or "admin" roles
+    And "repository" with "specified" name already created and stored in DB
+    When this user requested to save the same entity (key or value) which already exists with write permission for the relevant "repository" specified name
+    Then this new entity (key or value) shouldn't be created
+    And the user should get one of the error messages: "key: "specified" key name already exists" or "value": "specified" value name already exists"
```
Comment (aharonha) [inline scalecube-configuration/src/test/java/resources/features/mpa-7103-Automation-testing-for-the-configuration-servicen.feature]:
```suggestion
        Then a new value should be set to the given key and stored in the relevant "repository" of the DB
```

File: scalecube-configuration/src/test/java/resources/features/mpa-7103-Automation-testing-for-the-configuration-servicen.feature [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+    When this user requested to save the "specific" entity (key and value) with write permission for the relevant "repository" specified name
+    Then this new entity (key and value) shouldn't be created and the user should get an error message: "role: permission denied"
+
+
+  #MPA-7103 (#2.2)
+  Scenario: Fail to save (edit) the write permission for specific entity which already exist (duplicate)
+    Given a user have got a valid "token" (API key) with assigned "owner" or "admin" roles
+    And "repository" with "specified" name already created and stored in DB
+    When this user requested to save the same entity (key or value) which already exists with write permission for the relevant "repository" specified name
+    Then this new entity (key or value) shouldn't be created
```
Comment (aharonha) [inline scalecube-configuration/src/test/java/resources/features/mpa-7103-Automation-testing-for-the-configuration-servicen.feature]:
```suggestion
```

File: scalecube-configuration/src/test/java/resources/features/mpa-7103-Automation-testing-for-the-configuration-servicen.feature [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+    Given the user have been granted with valid "token" (API key) assigned by "member" role
+    And the relevant specified name "repository" was created and stored in DB
+    When this user requested to save the "instrumentInstance" entity in the relevant "repository" specified name with following details
+      | instrumentId | name | DecimalPrecision | Rounding | key                  |
+      | JPY          | Yen  | 4                | down     | KEY-FOR-CURRENCY-999 |
+    Then this new entity shouldn't be created and the user should get an error message: "role: permission denied"
+
+
+  #MPA-7103 (#5.2)
+  Scenario: Fail to save a duplicate of existent instrument instance entity
```
Comment (aharonha) [inline scalecube-configuration/src/test/java/resources/features/mpa-7103-Automation-testing-for-the-configuration-servicen.feature]: should not fail

**Review by aharonha — CHANGES_REQUESTED** [review]
do some changes

File: scalecube-configuration/src/test/java/resources/features/mpa-7103-Automation-testing-for-the-configuration-servicen.feature (hunk 4 above)
Comment (PavloEsco) [reply scalecube-configuration/src/test/java/resources/features/mpa-7103-Automation-testing-for-the-configuration-servicen.feature]: Agreed

File: scalecube-configuration/src/test/java/resources/features/mpa-7103-Automation-testing-for-the-configuration-servicen.feature [hunk 5]
```diff
@@ -62,42 +62,41 @@ Feature: Basic CRUD tests for configuration service.
...
 
 
   #MPA-7103 (#2.1)
-  Scenario: Fail to save (edit) the write permission for specific entity assigned by "Member" role
-    Given a user have got a valid "token" (API key) with assigned "member" role
-    And the relevant specified name "repository" was created and stored in DB
-    When this user requested to save the "specific" entity (key and value) with write permission for the relevant "repository" specified name
-    Then this new entity (key and value) shouldn't be created and the user should get an error message: "role: permission denied"
+  Scenario: Successful update (override - edit) the value for specific entity which already exist
+    Given a user have got a valid "tokens" (API key) with assigned "admin" and "owner" roles
```
Comment (aharonha) [inline scalecube-configuration/src/test/java/resources/features/mpa-7103-Automation-testing-for-the-configuration-servicen.feature:80]: token

Comment (aharonha) [inline scalecube-configuration/src/test/java/resources/features/mpa-7103-Automation-testing-for-the-configuration-servicen.feature:80]:
```suggestion
     Given a user have got a valid "token" (API key) with assigned "owner" role
```

File: scalecube-configuration/src/test/java/resources/features/mpa-7103-Automation-testing-for-the-configuration-servicen.feature [hunk 6]
```diff
@@ -62,42 +62,41 @@ Feature: Basic CRUD tests for configuration service.
...
+    And relevant "repositories" with "specified" names "Repo1" and "Repo2" already created and stored in DB with some "specific" entities
+    When this user requested to save (update) the these "specific" entities (key or value) in the relevant "repositories" name "Repo1" and "Repo2" applying the "admin" and "owner" tokens
```
Comment (aharonha) [inline scalecube-configuration/src/test/java/resources/features/mpa-7103-Automation-testing-for-the-configuration-servicen.feature:82]:
```suggestion
     When this user requested to update a "specific" value in the relevant "repositories" name "Repo1" and "Repo2" applying the "admin" and "owner" tokens
```

File: scalecube-configuration/src/test/java/resources/features/mpa-7103-Automation-testing-for-the-configuration-servicen.feature [hunk 7]
```diff
@@ -195,38 +206,54 @@ Feature: Basic CRUD tests for configuration service.
     When this user requested to save the "instrumentInstance" entity in the relevant "repository" name with following details
       | instrumentId | name   | DecimalPrecision | Rounding | key                        |
       | XAG          | Silver | 4                | down     | KEY-FOR-PRECIOUS-METAL-123 |
-    Then this new entity shouldn't be created and the user should get an error message like: "instrumentInstance" "instrumentId =  XAG already exists"
+    Then existent entity values shouldn't be duplicated thus new values should be set to the given keys and stored in the relevant "repositories" of the DB
```
Comment (aharonha) [inline scalecube-configuration/src/test/java/resources/features/mpa-7103-Automation-testing-for-the-configuration-servicen.feature:209]: `existent` -> `the existing`

**Review by aharonha — APPROVED** [review]

**Review by skolomiiets — APPROVED** [review]

**Pull request merged**